### ISSUE TITLE
BUG: Waterfox security notification system

### ISSUE THREAD

**Issue by nixiValor** [BrowserWorks/waterfox#17]
Hello, my name is vladimir

I think i have found a issue with **Waterfox security notification system**.

This [url](http://tinyurl.com/q8zqh6c) is an example url since its 350 chars long i have shortened it you can see the [preview url](http://preview.tinyurl.com/q8zqh6c) for its full length.

_This issue is the data uri base structure._

In Firefox latest version the url triggers this notification **"firefox prevented this site from asking you to install software on your computer."**

In Waterfox no notification is triggered.

**Expected result:**
    Waterfox should show security notification **"Waterfox prevented this site from asking you to install software on your computer."**

**Comment by MrAlex94** [comment]
Thanks for the report. At the moment both Firefox and Waterfox are behaving the same for me.

So the base64 encoded text that I can see is producing the following HTML code:

``` HTML
<html>
<head>
<script>
function injectURL(){
window.open("https://addons.mozilla.org/firefox/downloads/latest/10900/addon-10900-latest.xpi?src=cb-dl-mostpopular", '_self');
}
</script>
</head>
<body onload="injectURL();"></body>
</html>
```

To which both Firefox and Waterfox both just show a blank page: ![Screenshot](http://i.imgur.com/Ldc2WTX.jpg)

It looks like neither browsers are executing the function when it loads the page. Do you have a screenshot/recording of the results you're getting?

**Comment by nixiValor** [comment]
In Firefox 40.0.3 or higher it takes a few seconds to react but shows a notification.

![security-notification](https://cloud.githubusercontent.com/assets/14110775/9661608/c6ed4c18-529e-11e5-9522-d46450a8200f.jpg)

In watefox no notification is shown

**Comment by MrAlex94** [comment]
I've found the issue (it's fixed in 40.0.3). Could you test out [Waterfox 40.0.3](https://s3-eu-west-1.amazonaws.com/waterfox-download/releases/win64/installer/Waterfox+40.0.3+Setup.exe) to confirm it has been fixed?

**Comment by nixiValor** [comment]
@MrAlex94  yes the notification is showing correctly preventing installation.

**Comment by nixiValor** [comment]
is 40.0.3 getting released soon?

[use-after-free (& crash) after style flush in CanvasRenderingContext2D](https://bugzilla.mozilla.org/show_bug.cgi?id=1164766) part of
[Mozilla Firefox nsIPresShell Use-After-Free Remote Code Execution Vulnerability](https://bugzilla.mozilla.org/show_bug.cgi?id=1175278) was fixed on 40.0.3

Does waterfox take long to apply security updates or do you release them same or next day as mozilla.
